service/src/main/java/com/epam/esm/service/impl/TagServiceImpl.java

```diff
@@ -0,0 +1,93 @@
+package com.epam.esm.service.impl;
+
+import com.epam.esm.BaseDao;
+import com.epam.esm.converter.TagDTOToTagEntityConverter;
+import com.epam.esm.converter.TagToTagDTOConverter;
+import com.epam.esm.dto.TagDTO;
+import com.epam.esm.entity.Tag;
+import com.epam.esm.exception.DBCPDataSourceException;
+import com.epam.esm.exception.DaoException;
+import com.epam.esm.exception.ServiceException;
+import com.epam.esm.service.BaseService;
+import com.epam.esm.validator.PreparedValidatorChain;
+import com.epam.esm.validator.realisation.IntermediateTagLink;
+import com.epam.esm.validator.realisation.tag.TagNameValidatorLink;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class TagServiceImpl implements BaseService<TagDTO> {
+
+    @Autowired
+    private BaseDao<Tag> tagDao;
+
+    @Override
+    public List<TagDTO> findAll() throws ServiceException {
+        try {
+            List<Tag> listOfEntities = tagDao.findAll();
+            return listOfEntities.stream()
+                    .map(tag -> new TagToTagDTOConverter().apply(tag))
```

Comment:
New converter object on each operation. It will be a good idea to pass converter under the spring control.